src/main/java/lz/modelTesting/controllers/SurveysController.java

```diff
@@ -42,6 +45,18 @@ public void addSurvey(@RequestBody String postPayload) {
         surveysRepository.save(survey);
     }
 
+    @PostMapping(value = "/addMockupSurvey", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> addMockupSurvey(@RequestBody String postPayload) {
+        JSONObject jsonObject = new JSONObject(postPayload);
+        String name = jsonObject.getString(NAME);
+        String mockupId = jsonObject.getString(MOCKUP_ID);
+        String body = "{\"questions\":" + jsonObject.getJSONArray(BODY).toString() + '}';
+        Survey survey = new Survey(name, body);
+        survey.setMockupId(new JSONObject(postPayload).getString(MOCKUP_ID));
```

Comment:
* Maybe better create another constructor of survey with mockupId?
* you already have variable `mockupId`

Reply:
+ there already exists method `getSurveyFromJson`